pkg/requestcontext/context.go

```diff
@@ -63,10 +64,23 @@ func (acCtx *AccessControlContext) matchesRepo(globPatterns map[string]bool, rep
 
 // returns either a user has or not read rights on 'repository'.
 func (acCtx *AccessControlContext) CanReadRepo(repository string) bool {
-	return acCtx.matchesRepo(acCtx.ReadGlobPatterns, repository)
+	if acCtx.ReadGlobPatterns != nil {
+		return acCtx.matchesRepo(acCtx.ReadGlobPatterns, repository)
+	}
+
+	return true
 }
 
 // returns either a user has or not detectManifestCollision rights on 'repository'.
 func (acCtx *AccessControlContext) CanDetectManifestCollision(repository string) bool {
-	return acCtx.matchesRepo(acCtx.DmcGlobPatterns, repository)
+	if acCtx.DmcGlobPatterns != nil {
+		return acCtx.matchesRepo(acCtx.DmcGlobPatterns, repository)
+	}
+
+	return false
+}
+
+//nolint:unused
```

Comment:
Why is this unused? Shall we delete it in this case?

Reply:
I removed the unused code